packages/confirmation-modal/src/ConfirmationModal/ConfirmationModal.spec.tsx

```diff
@@ -238,90 +240,90 @@ describe('packages/confirmation-modal', () => {
       expect(confirmationButton).toHaveAttribute('aria-disabled', 'true');
     });
 
-    describe('resets the confirm button when the modal closes', () => {
-      test('on confirm', async () => {
-        const { findByTestId, getByLabelText, getByRole, rerender } =
-          renderModal({
-            open: true,
-            requiredInputText: 'Confirm',
-          });
-
-        const modal = getByRole('dialog');
-
-        const confirmationButton = await findByTestId(
-          LGIDS_CONFIRMATION_MODAL.confirm,
-        );
-        expect(confirmationButton).toHaveAttribute('aria-disabled', 'true');
-
-        let textInput = getByLabelText('Type "Confirm" to confirm your action');
-
-        fireEvent.change(textInput, { target: { value: 'Confirm' } });
-        expect(confirmationButton).not.toHaveAttribute('aria-disabled', 'true');
-
-        userEvent.click(confirmationButton as HTMLButtonElement);
-
-        await waitForElementToBeRemoved(modal);
-
-        rerender(
-          <ConfirmationModal
-            title="Title text"
-            open={true}
-            requiredInputText="Confirm"
-          >
-            Content text
-          </ConfirmationModal>,
-        );
-
-        textInput = getByLabelText('Type "Confirm" to confirm your action');
-
-        expect(textInput).toHaveValue('');
-        expect(confirmationButton).toHaveAttribute('aria-disabled', 'true');
-      });
-
-      test('on cancel', async () => {
-        const { findByTestId, getByLabelText, getByRole, rerender } =
-          renderModal({
-            open: true,
-            requiredInputText: 'Confirm',
-          });
-
-        const modal = getByRole('dialog');
-
-        const confirmationButton = await findByTestId(
-          LGIDS_CONFIRMATION_MODAL.confirm,
-        );
-        expect(confirmationButton).toHaveAttribute('aria-disabled', 'true');
-
-        const cancelButton = await findByTestId(
-          LGIDS_CONFIRMATION_MODAL.cancel,
-        );
-
-        let textInput = getByLabelText('Type "Confirm" to confirm your action');
-
-        fireEvent.change(textInput, { target: { value: 'Confirm' } });
-        expect(confirmationButton).not.toHaveAttribute('aria-disabled', 'true');
-
-        userEvent.click(cancelButton as HTMLButtonElement);
-
-        await waitForElementToBeRemoved(modal);
-
-        rerender(
-          <ConfirmationModal
-            title="Title text"
-            buttonText="Confirm"
-            open={true}
-            requiredInputText="Confirm"
-          >
-            Content text
-          </ConfirmationModal>,
-        );
-
-        textInput = getByLabelText('Type "Confirm" to confirm your action');
-
-        expect(textInput).toHaveValue('');
-        expect(confirmationButton).toHaveAttribute('aria-disabled', 'true');
-      });
-    });
+    const requiredInputTextCases = [
+      {
+        describeCase:
```

Comment:
In this case, we're testing that the confirm button is reset after the users types in the the required test, is that correct? 

We might not need this, but do we test that the disabled button remains disabled after closing the modal if the user does not type in the required text?

Reply:
> In this case, we're testing that the confirm button is reset after the user types in the required text, is that correct?
yes, I left those test cases as-is

> We might not need this, but do we test that the disabled button remains disabled after closing the modal if the user does not type in the required text?
I think the other test blocks that test that the button is enabled by typing in the `requiredInputText` should suffice